### ISSUE TITLE
[Feature Request?] OpenTelemetry

### ISSUE THREAD

**Issue by paychex-ssmithrand** [langchain-ai/langsmith-sdk#167]
Hi,

Are there any plans to utilize the OpenTelemetry SDK and APIs as the basis for generating observability Signals rather than a proprietary Tracing implementation?

**Comment by hinthornw** [comment]
We have an issue out for it - no concrete timeline yet though. Thank you for raising!

**Comment by hinthornw** [comment]
@paychex-ssmithrand Just as a followup. In your specific case, what would the ideal integration look like?  Something like a langsmith exporter?

**Comment by paychex-ssmithrand** [comment]
@hinthornw Good Q, let me bring that back to my team and I'll get back to you on that.

**Comment by hinthornw** [comment]
@paychex-ssmithrand LMK if you have more detailed requests

**Comment by paychex-ssmithrand** [comment]
@hinthornw Hey there,

So we were thinking of a more intrusive change to Langsmith in that the way it instruments Langchain uses OpenTelemetry under the hood.

Alternatvely, writing an [Automatic Instrumentation](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation) library for Langchain would also be helpful if Otel won't be used within Langsmith to generate traces, metrics, and logs. 

Hope that makes sense.

**Comment by hinthornw** [comment]
Got it. Thank you! I think it could be done via a callback handler / tracer in the langchain library

**Comment by paychex-ssmithrand** [comment]
@hinthornw Should I open a more specific request in the langchina repo?

**Comment by hinthornw** [comment]
Yes please! Opening an issue on the langchain repo may get additional community involvement. The core team is aware of the request but I can't promise an ETA right now
